workspace-manager-clienttests/src/main/java/scripts/testscripts/DataReferenceLifecycle.java

```diff
@@ -0,0 +1,53 @@
+package scripts.testscripts;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+import bio.terra.testrunner.runner.config.TestUserSpecification;
+import bio.terra.workspace.api.WorkspaceApi;
+import bio.terra.workspace.client.ApiException;
+import bio.terra.workspace.model.CloningInstructionsEnum;
+import bio.terra.workspace.model.CreateDataReferenceRequestBody;
+import bio.terra.workspace.model.DataReferenceDescription;
+import bio.terra.workspace.model.ReferenceTypeEnum;
+import scripts.utils.WorkspaceFixtureTestScriptBase;
+import scripts.utils.ClientTestUtils;
+
+public class DataReferenceLifecycle extends WorkspaceFixtureTestScriptBase {
+
+  @Override
+  public void doUserJourney(TestUserSpecification testUser, WorkspaceApi workspaceApi)
+      throws ApiException {
+    // Create a data reference
+    final CreateDataReferenceRequestBody body = ClientTestUtils.getTestCreateDataReferenceRequestBody();
+    final String referenceName = body.getName();
+
+    final DataReferenceDescription newReference = workspaceApi.createDataReference(body, getWorkspaceId());
+    assertDataReferenceDescription(newReference, referenceName);
+
+    // retrieve the data reference
+   final DataReferenceDescription retrievedById = workspaceApi.getDataReference(
+       getWorkspaceId(), newReference.getReferenceId());
+    assertDataReferenceDescription(retrievedById, referenceName);
+
+    // retrieve by name
+    final DataReferenceDescription retrievedByName = workspaceApi.getDataReferenceByName(
+        getWorkspaceId(), ReferenceTypeEnum.DATA_REPO_SNAPSHOT, referenceName);
+    assertDataReferenceDescription(retrievedByName, referenceName);
+
+    // remove reference
+    workspaceApi.deleteDataReference(getWorkspaceId(), newReference.getReferenceId());
+
+    // TODO: assert 404 when retrieving again. This may interfere with parent class checking of status.
```

Comment:
If you will merge this test with this loose end, then I'd suggest making a ticket for it and referencing the ticket in this comment.

Reply:
alternatively, we could remove the httpAssert calls from the base class in this PR instead of #183 and include the 404 assertion as part of this test. we should do one or the other, though.

Reply:
That's what I wound up doing.